app/main.py

```diff
@@ -1,3 +1,70 @@
-def shop_trip():
-    # write your code here
-    pass
+import datetime
+
+from app.car import Car
+from app.customer import Customer
+from app.shop import Shop
+from app import parsing
+
+
+def shop_trip() -> None:
+    config = parsing.parse_data_from_json()
+    customers_data = config["customers"]
+    shops_data = config["shops"]
+
+    for customer_data in customers_data:
+        car_data = customer_data.get("car")
+        if car_data:
+            car = Car(car_data["brand"], car_data["fuel_consumption"])
+        else:
+            car = Car("Default Car", 10.0)
+
+        customer = Customer(
+            customer_data["name"],
+            customer_data["location"],
+            customer_data["money"],
+            customer_data["product_cart"],
+            car
+        )
+
+        fuel_cost = customer.calculate_fuel_cost(customer.car,
+                                                 customer.location)
```

Comment:
`customer.location` wrong parameter for this function. Double check your function and which parameter it should take